src/actions/save/index.js

```diff
@@ -51,10 +50,22 @@ export const createSaveAction: SaveAction = (config: SaveActionConfig) => {
 
   return {
     type: "save",
-    save: () => {
+    save: (onError, paymentSourceDetails) => {
+      const { createVaultSetupToken } = config;
+    
+      if (!onError || !paymentSourceDetails) {
+        throw new ValidationError("Missing args to #save")
+      }
+
       return ZalgoPromise.try(() => {
-        // basics for typing requirements. Implementation to come in next ticket.
-      })
+        createVaultSetupToken()
+        .then((/* emptySetupToken */) => {
+          // take token and call our PP endpoint with it
+        }).catch((/* error */) => {
+          // TODO: Let's make sure we stringify this error safely/idiomatically - Do we define errors elsewhere?
+          return onError("Unable to retrieve setup token from 'createVaultSetupToken'")
```

Comment:
How can we make this simpler for merchants to parse?
Eg: we would want to send `errorCode: INTERNAL_SERVER`,`type:INVALID_TOKEN`&`errorMessage: createVaultSetupToken: Unable to retrieve setup token`. 
How do we pass this in the best possible manner?

Reply:
This is a great question. I'll raise that to the team in Slack.